packages/swarm-components/src/Dropdown.jsx

```diff
@@ -0,0 +1,462 @@
+/* eslint-disable */
+import React, {
+	createContext,
+	Children,
+	useContext,
+	useState,
+	useEffect,
+	useRef,
+} from 'react';
+import Portal from '@reach/portal';
+import Rect, { useRect } from '@reach/rect';
+import WindowSize from '@reach/window-size';
+import { node, func, object, string, number, oneOfType, any } from 'prop-types';
+import { wrapEvent, checkStyles, assignRef } from '@reach/utils';
+import { ForwardedButton } from './Button';
+
+// TODO: add the mousedown/drag/mouseup to select of native menus, will
+// also help w/ remove the menu button tooltip hide-flash.
+
+// TODO: add type-to-highlight like native menus
+
+let MenuContext = createContext();
+
+let checkIfAppManagedFocus = ({ refs, state }) => {
+	if (!state.isOpen && !!refs.menu) {
+		return !refs.menu.contains(document.activeElement);
+	}
+	return false;
+};
+
+let openAtFirstItem = state => ({ isOpen: true, selectionIndex: 0 });
+
+let close = state => ({
+	isOpen: true,
+	selectionIndex: -1,
+	closingWithClick: false,
+});
+
+let selectItemAtIndex = index => ({
+	selectionIndex: index,
+});
+
+let genId = prefix =>
```

Comment:
What is this for? Docs would be very helpful. It's also a one-liner function that is only used once, so I'd recommend keeping it inline in `getInitialMenuState` - no need for the `prefix` abstraction